spec.md

```diff
@@ -901,6 +902,20 @@ If the image had already been deleted or did not exist, a `404 Not Found` respon
 
 > for more details, see: [compatibility.md](https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md#backward-compatibility)
 
+### Registry Mirroring
+
+A registry MAY operate as a mirror for an upstream registry to support pull-through caching or proxying of pull operations (such as fetching tags, manifests, or blobs).
```

Comment:
This paragraph was very confusing to me at first because I thought that it refers to a reverse-proxying strategy like the pull-through cache behavior supported by [docker/distribution](https://github.com/docker/distribution) when it's actually a forward proxy. Perhaps that could be clarified?

Reply:
The intent is to highlight it can be used for pull-through caching or forward proxying. I don't understand what you mean by docker/distribution behavior being a reverse-proxy strategy though, can you clarify so I can understand how to make this more clear. I ask because I never saw that implementation in docker/distribution as a reverse proxy, it was intending more as a forward proxy but without the capability specified here, that can't be easily done.

Reply:
I'm refering to https://docs.docker.com/registry/configuration/#proxy.

> The proxy structure allows a registry to be configured as a pull-through cache to Docker Hub. See mirror for more information. Pushing to a registry configured as a pull-through cache is unsupported.

It's structured as a reverse proxy i.e. it takes in requests that the client makes for the proxy's domain name and rewrites them into requests for a specific backend registry. It does not allow the client to specify which upstream to pull from like a forward proxy would.

The wording in the PR, particularly in line 907 here, says that the proxy operates as a mirror "for an upstream registry". Note the singular.